python/ray/workers/setup_runtime_env.py

```diff
@@ -144,8 +188,33 @@ def current_ray_pip_specifier() -> Optional[str]:
         return f"ray[all]=={ray.__version__}"
 
 
-def inject_ray_and_python(conda_dict, ray_pip_specifier: Optional[str],
-                          py_version: str) -> None:
+def inject_ray_and_python(
+        conda_dict: Dict[Any, Any],
+        ray_pip_specifier: Optional[str],
+        py_version: str,
+        extra_pip_dependencies: Optional[List[str]] = None) -> Dict[Any, Any]:
+    """Add Ray, Python and (optionally) extra pip dependencies to a conda dict.
+
+    Args:
+        conda_dict (dict): A dict representing the JSON-serialized conda
+            environment YAML file.  This dict will be modified and returned.
+        ray_pip_specifier (str, optional): A string which can be passed to
+            `pip install` to install a certain Ray version.  This could be a
+            string "ray[all]", or a wheel URL.
+            If you don't want Ray to be injected into the dict, this parameter
+            should be set to None.
+        py_version (str): A string representing a Python version to inject
+            into the conda dependencies, e.g. "3.7.7"
+        extra_pip_dependencies (List[str]): A list of pip dependencies that
+            will be prepended to the list of pip dependencies in
+            the conda dict.  If the conda dict does not already have a "pip"
+            field, one will be created.
```

Comment:
I kind of feeling that these two overlaps.

Reply:
You're right, ray_pip_specifier is just another pip dependency.  Let me refactor this